pallets/attesters/src/lib.rs

```diff
@@ -27,81 +32,126 @@ pub mod pallet {
-    }
-
-    #[derive(Clone, Encode, Decode, Eq, PartialEq, Debug, TypeInfo)]
+    #[derive(Clone, Encode, Decode, Eq, PartialEq, Debug, TypeInfo, PartialOrd)]
     pub enum BatchStatus {
-        Pending,
-        ReadyForSubmission,
+        PendingMessage,
+        PendingAttestation,
+        ReadyForSubmissionByMajority,
+        ReadyForSubmissionFullyApproved,
         Committed,
     }
 
     #[derive(Clone, Encode, Decode, Eq, PartialEq, Debug, TypeInfo)]
-    pub struct Batch<Attester, Signature, BlockNumber> {
-        pub attestations: Vec<Attestation<Attester, Signature>>,
-        pub target: TargetId,
-        pub created: BlockNumber,
+    pub struct BatchMessage<BlockNumber> {
+        pub batch_sfx: Option<BatchConfirmedSfxId>,
+        pub next_committee: Option<CommitteeTransition>,
+        pub new_attesters: Option<AttestersChange>,
+        pub ban_attesters: Option<AttestersChange>,
+        pub remove_attesters: Option<AttestersChange>,
+        pub signatures: Vec<(u32, Signature65b)>,
         pub status: BatchStatus,
+        pub created: BlockNumber,
     }
 
-    impl<Attester, Signature, BlockNumber: Zero> Default for Batch<Attester, Signature, BlockNumber> {
-        fn default() -> Self {
-            Self {
-                attestations: Vec::new(),
-                target: [0u8; 4],
-                created: Zero::zero(),
-                status: BatchStatus::Pending,
+    // Add the following method to `BatchMessage` struct
+    impl<BlockNumber> BatchMessage<BlockNumber> {
+        pub fn message(&self) -> Vec<u8> {
+            let mut encoded_message = Vec::new();
+            if let Some(ref sfx) = self.batch_sfx {
+                sfx.encode_to(&mut encoded_message);
+                // Remove first 1 byte if the message is not empty to strip the SCALE-vector length prefix
+                if !encoded_message.is_empty() {
+                    encoded_message.remove(0);
+                }
+            }
+            if let Some(ref committee) = self.next_committee {
+                committee.encode_to(&mut encoded_message);
+            }
+            if let Some(ref attestors) = self.new_attesters {
+                attestors.encode_to(&mut encoded_message);
             }
+            if let Some(ref attestors) = self.ban_attesters {
+                attestors.encode_to(&mut encoded_message);
+            }
+            if let Some(ref attestors) = self.remove_attesters {
+                attestors.encode_to(&mut encoded_message);
+            }
+
+            encoded_message
         }
-    }
 
-    impl<Attester, Signature, BlockNumber> Batch<Attester, Signature, BlockNumber> {
-        pub fn new(target: TargetId, now: BlockNumber) -> Self {
-            Self {
-                attestations: Vec::new(),
-                target,
-                created: now,
-                status: BatchStatus::Pending,
+        pub fn message_hash(&self) -> H256 {
+            let mut keccak = Keccak::v256();
```

Comment:
will this always be a keccak hash? Even for non-ethereum messages?

Reply:
let's do only keccak for now, open a ticket for future implementations based on XDNS::ExecutionLayer

Reply:
opened issue for adjusting per different architectures https://github.com/t3rn/t3rn/issues/996